Curious failure for me on python 3.8:

```
builder for '/nix/store/cwcxigjmqxgrni8i3i3q5w8sfw2xlkp7-python3.8-fastavro-1.4.4.drv' failed with exit code 1; last 10 log lines:
    File "/nix/store/66p7i56lg3zzp7nj6g0fywb704ihp4iy-python3-3.8.11/lib/python3.8/contextlib.py", line 7, in <module>
      from types import MethodType
  ImportError: cannot import name 'MethodType' from partially initialized module 'types' (most likely due to a circular import) (/build/source/fastavro/types.py)
  =========================== short test summary info ============================
  FAILED tests/test_main_cli.py::test_cli_record_output - subprocess.CalledProc...
  FAILED tests/test_main_cli.py::test_cli_stream_input - subprocess.CalledProce...
  FAILED tests/test_main_cli.py::test_cli_arg_metadata - subprocess.CalledProce...
  FAILED tests/test_main_cli.py::test_cli_arg_schema - subprocess.CalledProcess...
  FAILED tests/test_main_cli.py::test_cli_arg_codecs - subprocess.CalledProcess...
  ============ 5 failed, 573 passed, 6 skipped, 1 deselected in 4.30s ============
```

--- a/pkgs/development/python-modules/fastavro/default.nix
+++ b/pkgs/development/python-modules/fastavro/default.nix
@@ -1,0 +1,59 @@
+{ buildPythonPackage
+, cython
+, fetchFromGitHub
+, isPy38
+, lib
+, lz4
+, numpy
+, pandas
+, pytestCheckHook
+, python-dateutil
+, python-snappy
+, pythonOlder
+, zstandard
+}:
+
+buildPythonPackage rec {
+  pname = "fastavro";
+  version = "1.4.4";
+
+  disabled = pythonOlder "3.6";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1sf8nqifwp0cggk59s22ygj3rm1nysa8b91xl8bpv2knqyjy1q32";
+  };
+
+  preBuild = ''
+    export FASTAVRO_USE_CYTHON=1
+  '';
+
+  nativeBuildInputs = [ cython ];
+
+  checkInputs = [
+    lz4
+    numpy
+    pandas
+    pytestCheckHook
+    python-dateutil
+    python-snappy
+    zstandard
+  ];
+
+  # Fails with "AttributeError: module 'fastavro._read_py' has no attribute
+  # 'CYTHON_MODULE'." Doesn't appear to be serious. See https://github.com/fastavro/fastavro/issues/112#issuecomment-387638676.
+  disabledTests = [ "test_cython_python" ];
+
+  # CLI tests are broken on Python 3.8. See https://github.com/fastavro/fastavro/issues/558.
+  disabledTestPaths = lib.optionals isPy38 [ "tests/test_main_cli.py" ];
+
+  pythonImportsCheck = [ "fastavro" ];
+
+  meta = with lib; {
+    description = "Fast read/write of AVRO files";
+    homepage = "https://github.com/fastavro/fastavro";
+    license = licenses.mit;
+    maintainers = with maintainers; [ samuela ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -2500,6 +2500,8 @@ in {
 
   fastapi = callPackage ../development/python-modules/fastapi { };
 
+  fastavro = callPackage ../development/python-modules/fastavro { };
+
   fastcache = callPackage ../development/python-modules/fastcache { };
 
   fastdiff = callPackage ../development/python-modules/fastdiff { };

